### DISCUSSION TITLE
URI on the frontend

### DISCUSSION THREAD

**Discussion by luka6000** [music-assistant/support#3243] [Feature requests and ideas]
Hello,

I’m the author of TagTuner player which works very well with MAss https://github.com/luka6000/TagTuner

I’m struggling explaining people how to get proper URIs for their playlists. I know how it works and search is kinda ok. But frankly speaking, search is pretty tough even for me. Especially having a few streaming sources (apple, yt, radio) and looking for this special version of the album/playlist.

So two feature requests:
1) having URI displayed directly in the ‚Provider details’ section next to original source
There was a similar feature request https://github.com/orgs/music-assistant/discussions/2311 but seams to be dropped?
2) possibility to play http/s links directly. It is quite easy to parse streaming url links to match proper provider and looking at URI it already has part of the source link in it. Why not play links directly? For example, Sonos works with direct links. Another point, streaming sources already have its addresses - why make it harder with new dedicated MAss only URIs?

**Comment by OzGav** [comment]
#2311 has not been dropped the item remains on the backlog but low priority

**Comment by DonGar** [comment]
Given that mass.search is no longer available in the official integration, perhaps we can raise the priority of #2311?

**Reply by OzGav** [reply]
mass.search is planned to be available in the core integration at the next HA update

**Comment by DonGar** [comment]
Excellent, thanks!

**Comment by OzGav** [comment]
In MA 2.4 the provider details section in the Artist, Album and Track views now have a chain link icon which will copy the URI to the clipboard. Note the browser must be accessing the MA frontend via HTTPS for this to work due to browser security rules.

**Reply by DonGar** [reply]
That is awesome, thanks!

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
